### ISSUE TITLE
Loss is very high even after many itearations

### ISSUE THREAD

**Issue by srika91** [martinkersner/train-CRF-RNN#12]
I have two classes people and background,when i tried to train for that,the loss still is a 5 digit number after 2000 iterations.How many iterations should I run for this dataset of people and background,is it usual for the loss to be so high even after 2000 iterations?The loss is stable around 85000 and occasionally  a 6 digit number.Am runnning in cpu so it is taking long time and hence asking here than waiting for more iterations

**Comment by martinkersner** [comment]
Hi @srika91!

First of all I would recommend you to use non-deep learning algorithm since you don't have access to GPU. It always takes time to get good results even with GPU.

The loss does not drop so fast, therefore you should keep training. However, if you read through other issues here, you would find that there is [some problem](https://github.com/martinkersner/train-CRF-RNN/issues/9) with training. Anyway, if you still want to use CRF-RNN you could expect to get good results after ten of thousands iterations. In a case you still want to use some kind of deep learning segmentation technique, you can try to train [DeepLab](https://github.com/martinkersner/train-DeepLab). It should converge faster. I get good results for PASCAL VOC dataset already after the first phase of training.

Martin

**Comment by srika91** [comment]
Thanks for the quick reply.I will certainly try the deeplab. I am looking to segment skin pixels,after segmenting human from the background.I have another dataset that is completely different from voc dataset related to document images.I have to do segmentation for different background conditions.Can I pretrain the deeplab models for completely different dataset?

**Comment by martinkersner** [comment]
I think that should work, keep weights from the first few layers and train the rest. And also I think it could be worth to try segmentation without presegmenting humans from background.

**Comment by srika91** [comment]
Why do you say its better to try without presegmentation?

**Comment by martinkersner** [comment]
I did't say it is better, I was just suggesting different approach. You would probably achieve better training and inference time, and results could stay same.

**Comment by srika91** [comment]
Oh kk thanks for the clarification.
